platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/DefaultConfiguration.java

```diff
@@ -0,0 +1,86 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.config;
+
+import com.swirlds.common.config.ConfigUtils;
+import com.swirlds.common.config.singleton.ConfigurationHolder;
+import com.swirlds.common.config.sources.LegacyFileConfigSource;
+import com.swirlds.common.config.sources.ThreadCountPropertyConfigSource;
+import com.swirlds.config.api.Configuration;
+import com.swirlds.config.api.ConfigurationBuilder;
+import com.swirlds.config.api.source.ConfigSource;
+import com.swirlds.platform.config.internal.ConfigMappings;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * A utility class for building a basic configuration with the default configuration sources and paths.
+ * <p>
+ * Can be used in cli tools to build a basic configuration.
+ */
+public class DefaultConfiguration {
+    private DefaultConfiguration() {
+        // Avoid instantiation for utility class
+    }
+
+    /**
+     * Build a basic configuration with the default configuration sources and paths.
+     * And register the configuration to the {@link ConfigurationHolder}.
+     *
+     * @return the configuration object
+     * @throws IOException if there is an error reading the configuration files
+     */
+    @NonNull
+    public static Configuration buildBasicConfiguration() throws IOException {
+        return buildBasicConfiguration(Collections.emptyList());
+    }
+
+    /**
+     * Build a basic configuration with the default configuration sources.
+     * And register the configuration to the {@link ConfigurationHolder}.
+     *
+     * @param configurationPaths additional paths to configuration files
+     * @return the configuration object
+     * @throws IOException if there is an error reading the configuration files
+     */
+    @NonNull
+    public static Configuration buildBasicConfiguration(@NonNull final List<Path> configurationPaths)
+            throws IOException {
+        final ConfigSource settingsConfigSource = LegacyFileConfigSource.ofSettingsFile();
+        final ConfigSource mappedSettingsConfigSource = ConfigMappings.addConfigMapping(settingsConfigSource);
+        final ConfigSource threadCountPropertyConfigSource = new ThreadCountPropertyConfigSource();
+
+        final ConfigurationBuilder configurationBuilder = ConfigurationBuilder.create()
+                .withSource(mappedSettingsConfigSource)
+                .withSource(threadCountPropertyConfigSource);
+        ConfigUtils.scanAndRegisterAllConfigTypes(configurationBuilder, Set.of("com.swirlds"));
+
+        for (final Path configurationPath : configurationPaths) {
+            System.out.printf("Loading configuration from %s%n", configurationPath);
```

Comment:
I'd recommend using a log4j2 logger here.

Reply:
PCLI uses system.out across the board, not normal loggers - I think this should not block this PR since this makes up a separate discussions about whether sysout is the right choice or not.